bing-dict-cache.el

```diff
@@ -0,0 +1,66 @@
+(defvar bing-dict-cache-auto-save t
+  "Whether to save the bing dict cache when leaving emacs.")
+
+(defvar bing-dict--cache nil
+  "It will be used when offline, its type is a hash table")
+
+(defvar bing-dict-cache-file (expand-file-name "var/bing-dict/bing-dict-save.el" user-emacs-directory)
+  "It is where cache will be stored")
+
+(defvar bing-dict-cache-limit 100
+  "The largest numbers of the items to be cached")
+
+(defun bing-dict--cache-init ()
+  "Initialize bing dict."
+  (setq bing-dict--cache (make-hash-table :test #'equal)))
+
+(defun bing-dict--cache-initialized-p ()
+  (hash-table-p bing-dict--cache))
+
+(defun bing-dict--cache-load ()
+  (load bing-dict-cache-file 'noerror nil 'nosuffix))
+
+(defun bing-dict--cache-overflow-p ()
+  (> (hash-table-count bing-dict--cache) bing-dict-cache-limit))
+
+(defun bing-dict--update-cache ()
```

Comment:
This function should be quite easy to test. It would be great to put some cache tests into `bing-dict-test.el`.

Reply:
Got it.